tests/functionaltests/UXTestHelpers.h

```diff
@@ -81,4 +84,22 @@ class XamlEventSubscription {
     StrongId<WXDependencyProperty> _propertyToObserve;
 }; // class XamlEventSubscription
 
+// Class used to signal and wait during a test run
+class UXEvent {
```

Comment:
>UXEvent [](start = 6, length = 7)

We'll want an autoreset mode too.  How about we make the ctor private and do:

public:

static std::shared_ptr<UXEvent> UXEvent::CreateManual(); {
    return std::make_shared<UXEvent>(false);
}

static  std::shared_ptr<UXEvent> UXEvent::CreateAuto() {
    return std::make_shared<UXEvent>(true);
}

..

private:

explicit UXEvent(bool autoReset);